version: '3'

services:
  syncer:
    init: true
    container_name: syncer
    hostname: syncer
    build: .
    depends_on:
      - database
    env_file:
      - .env
    restart: on-failure
    networks:
      botnet:
        ipv4_address: 10.10.13.3

  database:
    image: postgres:alpine
    container_name: postgres
    hostname: postgres
    restart: on-failure
    environment:
      POSTGRES_USER: syncer
      POSTGRES_DB: syncer-db
      POSTGRES_PASSWORD: ${DB_PG_PASSWORD}
    volumes:
      - ./deployment/scripts:/docker-entrypoint-initdb.d
    ports:
      - 5432:5432
      - 5433:5433
    networks:
      botnet:
        ipv4_address: 10.10.13.4

  pgadmin:
    container_name: pgadmin
    hostname: pgadmin
    image: dpage/pgadmin4
    restart: on-failure
    depends_on:
      - database
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@syncer.xyz
      PGADMIN_DEFAULT_PASSWORD: ${DB_PG_AD_PASSWORD}
    ports:
      - '5050:80'
    networks:
      botnet:
        ipv4_address: 10.10.13.5

  nginx:
    container_name: nginx
    hostname: nginx
    build:
      context: .
      dockerfile: ./deployment/nginx.Dockerfile
    ports:
      - '80:80'
    restart: always
    depends_on:
      - syncer
    networks:
      botnet:
        ipv4_address: 10.10.13.6

networks:
  botnet:
    driver: bridge
    ipam:
      config:
        - subnet: 10.10.13.0/24
          gateway: 10.10.13.1
